### PR TITLE
AJ-1257 Deleting a Row Referenced by Another Row in the Same Table

### PR DESCRIPTION
**Pull request by kevinmarete**
### What
- Fixes an issue where deleting an entity that is referenced by another entity in the same table fails silently. Now, if you try to delete such an entity, an error message will be shown: "In order to delete the selected data row, you must delete all references to the row in the same table."

### Why
- Previously, users received no feedback when a deletion failed due to internal references, which could be confusing. This change provides clear guidance on why the deletion cannot proceed.

### Testing strategy
- Manual Testing: Try deleting an entity with references within the same table and check for the error message.

https://github.com/user-attachments/assets/088cc0bc-b259-4a6e-80eb-bd88140f1395

https://github.com/user-attachments/assets/37f13f51-ae5c-402d-b6e1-a7f3b822ab66

### CODE REVIEW

File: src/workspace-data/data-table/entity-service/EntityDeleter.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import colors from 'src/libs/colors';
 import { reportError } from 'src/libs/error';
 import * as Utils from 'src/libs/utils';
 
-export const EntityDeleter = ({ onDismiss, onSuccess, namespace, name, selectedEntities, selectedDataType, runningSubmissionsCount }) => {
```
Comment (kevinmarete) [inline src/workspace-data/data-table/entity-service/EntityDeleter.js:10]: This is no longer used!

File: src/workspace-data/data-table/entity-service/EntityDeleter.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const EntityDeleter = ({ onDismiss, onSuccess, namespace, name, selectedE
   const doDelete = async () => {
     const entitiesToDelete = _.flow(
       _.map(({ name: entityName, entityType }) => ({ entityName, entityType })),
-      (entities) => _.concat(entities, additionalDeletions)
+      (entities) => _.concat(additionalDeletions, entities)
```
Comment (kevinmarete) [inline src/workspace-data/data-table/entity-service/EntityDeleter.js:19]: Begin by deleting the additional entities, followed by the selected entities.

File: src/workspace-data/data-table/entity-service/EntityDeleter.js [hunk 3]
```diff
@@ -27,11 +27,20 @@ export const EntityDeleter = ({ onDismiss, onSuccess, namespace, name, selectedE
...
+          setAdditionalDeletions(
+            _.filter(
+              (errorEntity) =>
+                !_.some(
+                  (selectedEntity) => selectedEntity.entityType === errorEntity.entityType && selectedEntity.entityName === errorEntity.entityName,
+                  entitiesToDelete
+                ),
+              await error.json()
+            )
+          );
```
Comment (kevinmarete) [inline src/workspace-data/data-table/entity-service/EntityDeleter.js:39]: Separate the selected entities from the additional entities to ensure the additional entities are deleted first, followed by the selected entities.

Comment (davidangb) [inline src/workspace-data/data-table/entity-service/EntityDeleter.js:39]: Optional, I won't block the PR on this feedback: the logic here to "find all entities in the json response which were not originally selected by the user" is somewhat complex. You could factor this out into a standalone function and write unit tests for it. I'm ok shipping as-is though.

**Review by davidangb — APPROVED** [review]
ty! one optional comment inline.

**Review by calypsomatic — APPROVED** [review]

**Pull request merged**